### DISCUSSION TITLE
Contributing / project status / community

### DISCUSSION THREAD

**Discussion by KG32** [yuankunzhang/charming#131] [General]
Hi, awesome crate. I use it for one of my projects and while it works great, there are some features I need that are not implemented yet though. This made me create a PR (https://github.com/yuankunzhang/charming/pull/130). I'd love to contribute more, but it got me thinking:
- is there a contributing guide planned? I couldn't really find anything in discussions either
- what's the project status? How often are new versions released?
- the Discord link is broken. Is there a plan to bring it back? Is taking care of that could be also a way to contribute?

Thanks,
Kuba

**Comment by LukaOber** [comment]
>is there a contributing guide planned? I couldn't really find anything in discussions either

There is no guide for contributing. Just open up a discussion, issue or PR, and we will manage from there. Are you missing something that you would like to have in a contributing guide?

>what's the project status? How often are new versions released?

There is no schedule to release new versions, if you are interested in the latest changes you can use the master branch. Breaking changes are very uncommon.

>the Discord link is broken. Is there a plan to bring it back? Is taking care of that could be also a way to contribute?

I am personally not a big fan of using Discord as a forum parallel to GitHub. I think, that for this project, all communication can happen in one place.

**Reply by KG32** [reply]
> There is no schedule to release new versions, if you are interested in the latest changes you can use the master branch. Breaking changes are very uncommon.

I see, why's that though? I can see there are over a month old changes merged. Sure, I can use main as a git source in dependencies if I really needed to, but
1. That's me, everyone else installing a crate are still missing them (like I did).
2. There's no guarantee there won't be breaking changes (even deleting or renaming this branch). There's no way I'd want to to that in production, or anyone else for that matter I believe.
3. What's the point and motivation to contribute if it's not published? Especially since it's a simple version bump and publish?

> I am personally not a big fan of using Discord as a forum parallel to GitHub. I think, that for this project, all communication can happen in one place.

Sure, with Discussions in use it makes sense, was simply wondering if that's intentional. Sounds like it'd be a good idea to remove it from readme then

**Reply by LukaOber** [reply]
> I see, why's that though? I can see there are over a month old changes merged. Sure, I can use main as a git source in dependencies if I really needed to, but
> 
>     1. That's me, everyone else installing a crate are still missing them (like I did).
> 
>     2. There's no guarantee there won't be breaking changes (even deleting or renaming this branch). There's no way I'd want to to that in production, or anyone else for that matter I believe.
> 
>     3. What's the point and motivation to contribute if it's not published? Especially since it's a simple version bump and publish?

I would like to make another release, see #110. But I am not the author and am unable to do so. The owner of this project is currently not actively working on this library. Which is completely fine, please don't pressure them into doing so. 

> Sure, with Discussions in use it makes sense, was simply wondering if that's intentional. Sounds like it'd be a good idea to remove it from readme then

You are of course correct, I updated the readme accordingly.
